ethereumj-core/src/main/java/org/ethereum/sync/ReceiptsDownloader.java

```diff
@@ -167,12 +166,12 @@ private void retrieveLoop() {
         while (!Thread.currentThread().isInterrupted()) {
             try {
 
-                if (toDownload.isEmpty()) {
-                    if (fillBlockQueue() > 0 || System.currentTimeMillis() - t > REQUEST_TIMEOUT) {
-                        toDownload = getToDownload();
-                        t = System.currentTimeMillis();
-                    }
+              if (toDownload.isEmpty()) {
+                if (fillBlockQueue() > 0 || System.currentTimeMillis() - t > REQUEST_TIMEOUT || toDownload.size() == 0) {
```

Comment:
`toDownload.size() == 0` is equal to `toDownload.isEmpty()` which is checked above, don't need to check it once again.